profiles/killbill/src/main/java/org/killbill/billing/server/filters/RequestDataFilter.java

```diff
@@ -31,22 +33,61 @@
 import org.glassfish.jersey.server.ContainerRequest;
 import org.glassfish.jersey.server.ContainerResponse;
 import org.glassfish.jersey.server.spi.ContainerResponseWriter;
+import org.killbill.billing.jaxrs.resources.JaxrsResource;
 import org.killbill.billing.util.UUIDs;
 import org.killbill.commons.request.Request;
 import org.killbill.commons.request.RequestData;
 
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
+
 @Singleton
 public class RequestDataFilter implements ContainerRequestFilter, ContainerResponseFilter {
 
     private static final String REQUEST_ID_HEADER = "X-Request-Id";
 
     private static final String LEGACY_REQUEST_ID_HEADER = "X-Killbill-Request-Id-Req";
 
+    private static final String KILL_BILL_ENCODING_HEADER = "X-Killbill-Encoding";
+
     @Override
     public void filter(final ContainerRequestContext requestContext) {
         final List<String> requestIdHeaderRequests = getRequestId(requestContext);
         final String requestId = (requestIdHeaderRequests == null || requestIdHeaderRequests.isEmpty()) ? UUIDs.randomUUID().toString() : requestIdHeaderRequests.get(0);
         Request.setPerThreadRequestData(new RequestData(requestId));
+
+        final List<String> kbEncodingHeaders = requestContext.getHeaders().get(KILL_BILL_ENCODING_HEADER);
+        final String kbEncodingHeader = (kbEncodingHeaders == null || kbEncodingHeaders.isEmpty()) ? null : kbEncodingHeaders.get(0);
+        if (kbEncodingHeader != null) {
```

Comment:
Should not we check the value here, i.e 'Base64' ?

Reply:
It's done line 73: https://github.com/killbill/killbill/pull/1573/files/404496a97513d3f004998eeabc2e6221a077a097#diff-0f99ec375e3ac23eab1c6990a24a66790c9ebeaa27f242d4317883edced53eedR73

Reply:
Yep, my point is that you probably did not need to do it each, just one time here would have been enough.

Reply:
Ah, right -- I was anticipating the case where we support multiple schemes down the line (so need to find the right decoder each time) 😄